src/Microsoft.OData.Core/SelectedPropertiesNode.cs

```diff
@@ -117,10 +117,23 @@ private void Parse(string selectClause)
                 this.ParsePathSegment(segments, 0);
             }
 
-            if ( (this.selectedProperties == null || this.selectedProperties.Count == 0)
-                && this.children.Values.All( n => n.selectionType == SelectionType.EntireSubtree))
+            DetermineSelectionType(this);
+        }
+
+        private static void DetermineSelectionType(SelectedPropertiesNode node)

```

Comment:
>SelectedPropertiesNode node [](start = 51, length = 27)

It seems we also can use instance method, not the static. But, either way is ok for me. Only to call this static method using "this" a little bit confused me.

Reply:
Yes, I like that better.

---
In reply to: [242370658](https://github.com/OData/odata.net/pull/1359#discussion_r242370658) [](ancestors = 242370658)